homeassistant/components/hue/light.py

```diff
@@ -145,16 +157,21 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
         ),
     )
 
-    update_groups = partial(
-        async_update_items,
-        bridge,
-        bridge.api.groups,
-        {},
-        async_add_entities,
-        partial(create_light, HueLight, group_coordinator, bridge, True),
-    )
+    update_rooms = partial(async_update_rooms, bridge.api.groups, rooms)
+
+    if allow_groups:
+        update_groups = partial(
+            async_update_items,
+            bridge,
+            bridge.api.groups,
+            {},
+            async_add_entities,
+            partial(create_light, HueLight, group_coordinator, bridge, True, None),
+        )
+
+        bridge.reset_jobs.append(group_coordinator.async_add_listener(update_groups))
 
-    bridge.reset_jobs.append(group_coordinator.async_add_listener(update_groups))
+    bridge.reset_jobs.append(group_coordinator.async_add_listener(update_rooms))
```

Comment:
Should we listen to the groups all the time? We only need it when an entity is added to HA rihgt? It means that even if no group entities are created, we are still going to fetch this to stay up to date.

Reply:
I'd have to teach `async_update_items` about the relationship between groups and lights as well as store some data elsewhere since its all in the closure later.

Doable for sure, just adds a lot of complexity.

Reply:
Hmm.. If I do that and they move the devices around it won't pickup the room change.. not sure thats actually a problem though

Reply:
That's not a problem because once we suggest we don't unsuggest.

Reply:
I've implemented that. it cancels the listener after each room update and re-adds it when a new light is added